packages/server/src/ApolloServer.ts

```diff
@@ -138,6 +138,7 @@ export interface ApolloServerInternals<TContext extends BaseContext> {
   formatError?: (
     formattedError: GraphQLFormattedError,
     error: unknown,
+    context?: TContext,
```

Comment:
Please change the name of all symbols in the PR that are of type `TContext` from `context` to `contextValue`. In AS4 we are trying to use naming to help distinguish between a context value and a context function.

I think a similar declaration in `constructor.ts` needs to change too. In fact, you could change this declaration to be `formatError?: ApolloServerOptionsBase<TContext>['formatError']` if you'd like.

I think it would also be better to make this be `contextValue: TContext | undefined`, and explicitly pass in the `undefined` in the case where it's not there (the case from `errorResponse` in this file where the error happens before the context is successfully created). Let's make sure every call to `normalizeAndFormatErrors` has to explicitly decide whether it's passing a `contextValue` or not. Additionally, there's at least one call to `errorResponse` (the one at the very end of `executeHTTPGraphQLRequest` that's basically "something surprising threw") where we actually might have a contextValue, so let's not lose it in that case...